### ISSUE TITLE
AKS with VMSS

### ISSUE THREAD

**Issue by epa095** [terraform-providers/terraform-provider-azurerm#2900]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

`az` can create aks clusters with virtual scale sets enabled by  `--enable-vmss`, and we would like to be able to do the same in terraform. VMSS is a requirement for node autoscaling, which is cool. 

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

**Comment by Lucretius** [comment]
It looks like the new version of the SDK contains the necessary autoscaling configuration properties so this  can be done with a bump to the containerservice API version being used.

**Comment by tombuildsstuff** [comment]
@Lucretius the new version of the SDK being the SDK version or the API Version?

**Comment by Lucretius** [comment]
Sorry, the new version of the API within the existing SDK.  See [here](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice)

**Comment by tombuildsstuff** [comment]
@Lucretius looks like this would require bumping the Azure SDK too, we're currently using `v25.1.0` where that API version isn't available: https://github.com/Azure/azure-sdk-for-go/tree/v25.1.0/services/containerservice/mgmt

**Comment by djsly** [comment]
I was looking at the same thing. 

https://github.com/Azure/azure-sdk-for-go/tree/v26.5.0/services/containerservice/mgmt/2019-02-01/containerservice

comes with >= v26.5.0 

@tombuildsstuff  do you need to track the bump of the sdk in a separate issues / PR

**Comment by tombuildsstuff** [comment]
@djsly sorry I meant to open a PR for the SDK upgrade last week (but it dropped off my radar) - let me do that now

**Comment by tombuildsstuff** [comment]
@djsly PR to update the SDK: https://github.com/terraform-providers/terraform-provider-azurerm/pull/3126

**Comment by landro** [comment]
Please also remember adding availability zone support while enabling vmss.

Availability zones are currently supported in the go SDK

```go
	// AvailabilityZones - (PREVIEW) Availability zones for nodes. Must use VirtualMachineScaleSets AgentPoolType.
AvailabilityZones *[]string `json:"availabilityZones,omitempty"`
```

### References 
https://github.com/Azure/azure-sdk-for-go/blob/master/services/containerservice/mgmt/2019-04-30/containerservice/models.go

https://github.com/Azure/azure-cli-extensions/tree/master/src/aks-preview#enable-vmss-for-new-cluster-with-availability-zone-feature

**Comment by epa095** [comment]
I guess this can be closed now that #3422 is merged @dominik-lekse?

**Comment by dominik-lekse** [comment]
@epa095 @tombuildsstuff Yes, this feature is in the latest 1.28.0 release and the issue can be closed.
